lib/jsdom/level2/style.js

```diff
@@ -173,6 +173,13 @@ core.HTMLStyleElement._init = function() {
 
     evaluateStylesheet.call(this, content, this.sheet, this._ownerDocument.URL);
   });
+
+  this.addEventListener('DOMNodeRemovedFromDocument', function() {
```

Comment:
We're trying not to use mutation events going forward, since they are no longer spec compliant, and should be removed from jsdom. Instead of creating a new event listener for every stylesheet, maybe you could add a _descendantRemoved handler to Document?